### DISCUSSION TITLE
How to send OTP and not Magic Link to `supabase.auth.signInWithOtp`

### DISCUSSION THREAD

**Discussion by herbievine** [supabase/supabase#17581] [Questions]
In the docs it's stated that you can choose whether to receive a Magic Link or a OTP, but I don't see the options in the types...

https://supabase.com/docs/reference/javascript/auth-signinwithotp

Types:

```
export type SignInWithPasswordlessCredentials =
  | {
      email: string
      options?: {
        emailRedirectTo?: string
        shouldCreateUser?: boolean
        data?: object
        captchaToken?: string
        // No other options?
      }
    }
  | {
      phone: string
      options?: { ... }
    }
```

**Comment by GaryAustin1** [comment]
You put {{.Token}} in the template and remove the {{ .ConfirmationURL }} part.

**Comment by tomelliot** [comment]
~I'm looking at switching to OTP for logins. I'm not seeing a way to do this with local development. Is there somewhere that I can configure the emails for local development?~

Info on setting up emails for local development is here: 
https://supabase.com/docs/guides/cli/customizing-email-templates
